pkg/playback/http_playback.go

```diff
@@ -0,0 +1,660 @@
+	if err != nil {
+		return &http.Response{}, err
+	}
+
+	response := (*responseWrapper).(*http.Response)
+
+	return response, err
+}
+
+// Reads any contiguous set of webhook recordings from the start of the cassette
+// The caller must call response.Body().Close() on each of the returned http.Response's, otherwise there will be a resource leak
+func (httpReplayer *replayServer) readAnyPendingWebhookRecordingsFromCassette() (webhookRequests []*http.Request, webhookResponses []*http.Response, err error) {
+	webhookBytes := make([]cassettePair, 0)
+
+	// --- Read the pending webhook interactions (stored as raw bytes) from the cassette
+	for httpReplayer.replayer.interactionsRemaining() > 0 {
+		interaction, err := httpReplayer.replayer.peekFront()
+		if err != nil {
+			return nil, nil, fmt.Errorf("Error when checking webhooks: %w", err)
+		}
+
+		if interaction.Type == incomingInteraction {
+			webhookBytes = append(webhookBytes, interaction)
+			_, err = httpReplayer.replayer.popFront()
+			if err != nil {
+				return nil, nil, fmt.Errorf("Unexpectedly reached end of cassette when checking for pending webhooks: %w", err)
+			}
+		} else {
+			break
+		}
+	}
+
+	// --- Deserialize the bytes into HTTP request & response pairs
+	webhookRequests = make([]*http.Request, 0)
+	webhookResponses = make([]*http.Response, 0)
+	for _, rawWebhookBytes := range webhookBytes {
+		// TODO(bwang): having to create a new instance to get at the fromBytes() method doesn't feel great.
+		// But is it unavoidable since Golang doesn't have 'static' methods? Is there some other refactoring we can do?
+		requestSerializer := newSerializableHTTPRequest(&http.Request{})
+		var reqReader io.Reader = bytes.NewReader(rawWebhookBytes.Request)
+		webhookHTTPRequest, err := requestSerializer.fromBytes(&reqReader)
+		if err != nil {
+			return nil, nil, fmt.Errorf("Error when deserializing cassette to replay webhooks: %w", err)
+		}
+
+		webhookRequests = append(webhookRequests, webhookHTTPRequest.(*http.Request))
+
+		responseSerializer := newSerializableHTTPResponse(&http.Response{})
+		var respReader io.Reader = bytes.NewReader(rawWebhookBytes.Response)
+		webhookHTTPResponse, err := responseSerializer.fromBytes(&respReader)
+		if err != nil {
+			return nil, nil, fmt.Errorf("Error when deserializing cassette to replay webhooks: %w", err)
+		}
+
+		// Caller of this function is expected to call .Body().Close() on each http.Response
+		webhookResponses = append(webhookResponses, webhookHTTPResponse.(*http.Response)) // nolint
+	}
+
+	return webhookRequests, webhookResponses, nil
+}
+
+func (httpReplayer *replayServer) initializeServer(address string) *http.Server {
+	customMux := http.NewServeMux()
+	customMux.HandleFunc("/", httpReplayer.handler)
+
+	server := &http.Server{Addr: address, Handler: customMux}
+
+	return server
+}
+
+// These constants define the modes the playback server can be in
```

Comment:
I'd move everything below this line to a separate file

Reply:
How about this reorganization?

- `struct ReplayServer` -> `http_replayer.go`
- `struct RecordServer` -> `http_recorder.go`
- rename `RecordReplayServer` -> `PlaybackServer`
- `struct PlaybackServer` -> (stays in) `http_playback.go`

And also, rename `core_playback.go` -> `interactions_core.go` (to distance it from the naming of `http_playback.go`)